crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -107,7 +107,11 @@ impl C8yEndPoint {
 
         let (_, host) = url_host.split_once('.').unwrap_or((url_host, ""));
         let (_, c8y_host) = tenant_uri.split_once('.').unwrap();
-        (host == c8y_host).then_some(url)
+
+        // The configured `c8y.http` setting may have a port value specified,
+        // but the incoming URL is less likely to have any port specified.
+        // Hence just matching the host prefix.
```

Comment:
The other option was to change the `C8yEndPoint::c8y_host` field into a `HostPort` and just validate the `host` part of it against the incoming host. But, that would be have been a huge change needing the generic port parameter `<P>` to be added to `C8yEndPoint` as well, since `HostPort<P>` is gene]
ric.